include/ignition/gui/qml/PlottingInterface.qml

```diff
@@ -0,0 +1,250 @@
+                  {
+                      // skip the first one
+                      if (firstChart)
+                      {
+                          firstChart = false;
+                          // make that chart has a full size
+                          chartsLayout.heightFactor = 1
+                          continue;
+                      }
+                      var chart = chartsLayout.children[i];
+                      reallocateChart(chart);
+                  }
+              } // end if
+
+              addChart();
+          }
+      }
+  }
+
+  function addChart()
+  {
+      main.idIncrementor++;
+
+      var chartComponent = Qt.createComponent("Chart.qml");
+      var chartObject;
+      // if the mode is many charts that are organized in horizontal layout
+      if (multiChartsMode)
+      {
+          // add it to the horizontal layout
+          chartObject = chartComponent.createObject(rowChartsLayout);
+          chartObject.width = 200;
+          chartObject.height = Qt.binding( function() {return rowCharts.height * 0.9} );
+          chartObject.y = Qt.binding( function() {return (rowCharts.height - chartObject.height)/2} );
+          chartObject.multiChartsMode = true;
+          chartObject.fillPlotInOrOut()
+      }
+      else
+      {
+          // if normal mode, add the chart to the normal vertical layout
+          chartObject = chartComponent.createObject(chartsLayout);
+          chartObject.height = Qt.binding( function() {return chartsLayout.height / chartsLayout.heightFactor});
+          chartObject.width = Qt.binding( function() {return chartsLayout.width});
+          // to change the height of each vertical chart
+          chartsLayout.heightFactor ++ ;
+      }
+
+      // Chart ID
+      chartObject.chartID = main.idIncrementor;
+      charts[idIncrementor] = chartObject;
+      // Signals and Slots
+      chartObject.subscribe.connect(main.onSubscribe);
+      chartObject.unSubscribe.connect(main.onUnSubscribe);
+      chartObject.componentSubscribe.connect(main.onComponentSubscribe);
+      chartObject.componentUnSubscribe.connect(main.onComponentUnSubscribe);
+      chartObject.clicked.connect(main.onClicked);
+  }
+
+  function reallocateChart(chart)
+  {
+      // new attributes to fit in the horizontal layout
+      chart.width = 200;
+      chart.height = Qt.binding( function() {return rowCharts.height * 0.9} );
+      chart.y = Qt.binding( function() {return (rowCharts.height - chart.height)/2} );
+      // enable that to enable the some UI features in the chart
+      chart.multiChartsMode = true;
+      chart.fillPlotInOrOut();
+
+      // change the layout from the vertical layout to the top horizontal one
+      chart.parent = rowChartsLayout;
+  }
+
+  function onSubscribe(Id, topic, path)
+  {
+      PlottingIface.subscribe(topic, path, Id);
+  }
+  function onUnSubscribe(Id, topic, path)
+  {
+      PlottingIface.unsubscribe(topic, path, Id);
```

Comment:
If we really need to forward these calls, can we order the parameters the same way on both of them?